compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClassClosureDesugar.java

```diff
@@ -919,9 +919,9 @@ public void visit(BLangStringTemplateLiteral stringTemplateLiteral) {
     }
 
     @Override
-    public void visit(BLangWorkerSend workerSendNode) {
-        workerSendNode.expr = rewriteExpr(workerSendNode.expr);
-        result = workerSendNode;
+    public void visit(BLangWorkerAsyncSendExpr asyncSendExpr) {
+        asyncSendExpr.expr = rewriteExpr(asyncSendExpr.expr);
+        result = asyncSendExpr;
```

Comment:
Closures seem not identified properly due to this issue https://github.com/ballerina-platform/ballerina-lang/issues/38810